### PR TITLE
root: migrate to python@3.11

### PR DESCRIPTION
**Pull request by ryenus**
Update formula **root** to use python@3.11 instead of python@3.10, see the related pr https://github.com/Homebrew/homebrew-core/pull/114154

### CODE REVIEW

**Comment by chenrui333** [comment]
```
/tmp/root-20221226-40023-15av6hk/root-6.26.06/bindings/pyroot/cppyy/CPyCppyy/src/CPPOverload.cxx:5:10: fatal error: code.h: No such file or directory
    5 | #include "code.h"            // from Python
      |          ^~~~~~~~
compilation terminated.
```

**Comment by cho-m** [comment]
Backport PR for Python 3.11 support into 6.26: https://github.com/root-project/root/pull/11733.

May also need to update to 6.26.10 (#114543) which also mentions Python 3.11 changes.

**Comment by ryenus** [comment]
Not stale

**Comment by cclauss** [comment]
Line 4:
> url "https://root.cern.ch/download/root_v6.26.06.source.tar.gz"
-->
> url "https://root.cern.ch/download/root_v6.26.10.source.tar.gz"

**Comment by ryenus** [comment]
Related pr #114543 - [root minuit2 6.26.10](https://github.com/Homebrew/homebrew-core/pull/114543)

**Comment by ryenus** [comment]
Hmm, @cho-m already mentioned that [above](https://github.com/Homebrew/homebrew-core/pull/119098#issuecomment-1367084514). I agree it's better to converge the 2 pr(s).

File: Formula/root.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 class Root < Formula
   desc "Object oriented framework for large scale data analysis"
   homepage "https://root.cern.ch/"
-  url "https://root.cern.ch/download/root_v6.26.06.source.tar.gz"
+  url "https://root.cern.ch/download/root_v6.26.10.source.tar.gz"
```
Comment (cclauss) [inline Formula/root.rb:4]: https://root.cern.ch/download/root_v6.28.00.source.tar.gz

File: Formula/root.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
   sha256 "b1f73c976a580a5c56c8c8a0152582a1dfc560b4dd80e1b7545237b65e6c89cb"
```
Comment (cclauss) [inline Formula/root.rb:5]:
```suggestion
  url "https://root.cern.ch/download/root_v6.28.00.source.tar.gz"
  sha256 "afa1c5c06d0915411cb9492e474ea9ab12b09961a358e7e559013ed63b5d8084"
```

**Comment by ryenus** [comment]
Closing since this pr is effectively a duplicate of https://github.com/Homebrew/homebrew-core/pull/114543

**Pull request closed without merging**